frontend-react/src/pages/submissions/SubmissionDetails.tsx

```diff
@@ -0,0 +1,182 @@
+import { Suspense } from 'react';
+import { useParams } from 'react-router-dom';
+import { NetworkErrorBoundary, useResource } from 'rest-hooks';
+import { getStoredOrg } from '../../components/GlobalContextProvider';
+import Spinner from '../../components/Spinner';
+import Title from '../../components/Title';
+import ActionDetailsResource from '../../resources/ActionDetailsResource';
+import { Destination } from '../../types/SubmissionDetailsTypes'
+import { ErrorPage } from '../error/ErrorPage';
+
+/* Custom types */
+type DetailItemProps = {
+    item: string,
+    content: any,
+    subItem?: boolean;
+}
+
+type DestinationItemProps = {
+    destinationObj: Destination
+}
+
+type SubmissionDetailsProps = {
+    actionId: string | undefined
+}
+
+type SubmissionDate = {
+    dateString: string;
+    timeString: string;
+}
+
+/* 
+    This function serves as a cleaner (read: more contained) way of parsing out
+    necessary date and time string formats for this page.
+
+    @param dateTimeString - the value representing when a report was sent, returned
+    by the API  
+    
+    @returns SubmissionDate
+    dateString format: 1 Jan 2022
+    timeString format: 3:00 PM
+*/
+// TODO: Refactor, maybe??
+export const generateSubmissionDate = (dateTimeString: string): SubmissionDate => {
```

Comment:
I'm glad you're NOT using `moments`... that's a very heavy weight library to do just date/time formatting.

We should definitely put this somewhere common and have all places use the same format. (e.g. remove `moment` calls from other locations.)

You might want to do some safety checking? 
`generateSubmissionDate('')` > "the batman song" aka (`{dateString: "NaN undefined NaN", timeString: "NaN:NaN AM"}`

Discussion (no change required):
Hard coding months (and things "am" and "pm") in an array like this leads to localization issues later. Not sure how much it matters for this project, but good to keep in mind.

Typically, it's safest/boring to just use toLocaleString() with some options.

```js
const twosDay= "Feb 22, 2022 2:22 pm";

generateSubmissionDate(twosDate) 
 > {dateString: "22 Feb 2022", timeString: "2:22 PM"}

new Date(twosDay).toLocaleString() 
  > 2/22/2022, 2:22:00 PM

new Date(twosDay).toLocaleString('en-US', {
  timeZoneName: 'short',
}) 
 > '2/22/2022, 2:22:00 PM PST'

new Date(twosDay).toLocaleString('en-GB', {
  timeZoneName: 'short',
}) 
 > '22/02/2022, 14:22:00 GMT-8'
```

Reply:
Definitely something I considered but ultimately since this was taking a UTC datetime string from the server and parsing it to display a time, it felt safe. I'll add a disclaimer in the notes that this is purely for parsing UTC datetimes from the server, **not** for generating a truthful datetime.